src/DynamicFormsProvider.php

```diff
@@ -66,14 +70,15 @@ public function boot()
 
         /** @var ComponentRegistry $registry */
         $registry = $this->app->make(ComponentRegistry::class);
-
+        /** @var ResourceRegistry $resourceRegistry */
+        $resourceRegistry = $this->app->make(ResourceRegistry::class);
         /** @var JSONLogicInitHelper $jsonHelper */
         $jsonHelper = $this->app->make(JSONLogicInitHelper::class);
         /** @var FileComponentRegistry $fileRegistry */
         $fileRegistry = $this->app->make(FileComponentRegistry::class);
 
-        Request::macro('validateDynamicForm', function (string $definitionJson, string $submissionJson) {
-            $formDefinition = new Form($definitionJson); // @TODO pass registry
+        Request::macro('validateDynamicForm', function (string $definitionJson, string $submissionJson) use ($resourceRegistry) {
```

Comment:
This should *default* to making the resource registry from the container, but allow it to be passed. 

That way, in CAP, the controller can look at what org the submission is for and adjust the registry accordingly (no org overlap). But most other users of dynamic-forms probably won't care to do that, so defaulting it means they don't need to think about it (unless they need it).

So move that to be a third parameter that's `?ResourceRegistry $resourceRegistry` and if it's null use the helpers[^1] to make it from the container (`app()->make()` or just `resolve()`).

[^1]:  So it gets the correctly-sandboxed app container when run on Octane.